ergo/metaculus.py

```diff
@@ -698,67 +703,61 @@ def show_prediction(
-                }
-            )
-            # get domain for graph given the percentage of distribution kept
-            _xmin, _xmax = self.denormalize_samples(
-                self.get_central_quantiles(
-                    prediction_normed_samples,
-                    percent_kept=percent_kept,
-                    side_cut_from=side_cut_from,
-                )
-            )
 
-            df["prediction"] = self.denormalize_samples(df["prediction"])
-            df["community"] = self.denormalize_samples(df["community"])
+            if type(samples) == pd.DataFrame:
+                if plot != "samples" and samples.shape[1] > 1:
+                    raise ValueError(
+                        "For multiple predictions comparisons, only samples can be compared (The plot parameter can only be 'samples')"
+                    )
+                for col in samples:
+                    df[col] = self.normalize_samples(samples[col])
+            else:
+                df["samples"] = self.normalize_samples(samples)
 
-            df = pd.melt(df, var_name="sources", value_name="samples")  # type: ignore
+        if plot in ("fitted", "both"):
+            prediction = self.get_submission_from_samples(samples)
+            df["fitted"] = pd.Series(
+                [logistic.sample_mixture(prediction) for _ in range(0, num_samples)]
+            )
 
-            plot = self.comparison_plot(df, _xmin, _xmax, **kwargs)
-            plot.draw()
+        if show_community:
+            df["community"] = [  # type: ignore
+                self.sample_normalized_community() for _ in range(0, num_samples)
+            ]
 
-        else:
-            df = pd.DataFrame(data={"prediction": prediction_normed_samples})  # type: ignore
-            # get domain for graph given the percentage of distribution kept
-            _xmin, _xmax = self.denormalize_samples(
-                self.get_central_quantiles(
-                    df, percent_kept=percent_kept, side_cut_from=side_cut_from
-                )
+        # get domain for graph given the percentage of distribution kept
+        xmin, xmax = self.denormalize_samples(
+            self.get_central_quantiles(
+                df, percent_kept=percent_kept, side_cut_from=side_cut_from,
             )
+        )
 
-            df["prediction"] = self.denormalize_samples(df["prediction"])
+        for col in df:
+            df[col] = self.denormalize_samples(df[col])
 
-            plot = self.density_plot(df, _xmin, _xmax, fill="#b3cde3", **kwargs) + labs(
-                x="Prediction", y="Density", title=self.plot_title
-            )
-            plot.draw()
+        df = pd.melt(df, var_name="sources", value_name="samples")  # type: ignore
+
+        plot = self.comparison_plot(df, xmin, xmax, **kwargs) + labs(
+            x="Prediction",
+            y="Density",
+            title=self.plot_title + "\n\nPrediction vs Community"
```

Comment:
I think this fixes an "API bug" where comparison plots sometimes had their titles left off

Reply:
Yep, the change in question.name behavior broke a couple unexpected things (and I didn't know that the behavior was changed :) ). That is why it is important to have tests I reckon

Reply:
haha yeah :)